_posts/2020-05-19-hhvm-4.58.markdown

```diff
@@ -0,0 +1,59 @@
+---
+title: "HHVM 4.58"
+layout: post
+author: fred 
+category: blog
+---
+
+HHVM 4.58 is released! This release marks the end of support for 4.52;
+HHVM 4.53&ndash;4.57 remain supported, as does the 4.32 LTS release.
+
+# Highlights
+
+- Errors for cyclic definitions (e.g. when using type aliases) are no longer
+  duplicated, or raised in files that merely use a problematic type.
+- If an error is raised for type constant in a superclass, duplicate errors are
+  no longer raised for subclasses.
+- When trait methods are incorrectly overridden, the typechecker now reports
+  errors at the override, instead of at the original trait definition.
+- Added `.hackpartial` files, to replace `<?hh // partial`; like `.hack` files,
+  these **must not** contain a header line. See 'future changes' for upcoming
+  breaking changes related to partial files.
+- Fixed issues passing FDs to child processes via `proc_open` when using
+  LightProcess, especially when `STDIN` was not specified, or other low-numbers
+  FDs were specified.
+
+# Breaking Changes
+
+- Support for soft type hints using `@` (e.g. `@int $foo`) has been disabled;
+  use `<<__Soft>> int $foo` instead.
+- Static closures (e.g. `$f = static function() { return 1;}`) has been disabled
+  as they were never safely typechecked.
+- `goto` is now a parse error; it has never been supported in Hack code.
```

Comment:
Strictly speaking, we've just changed the default options for the parser. You can still use soft type hints if you set `disable_legacy_soft_typehint = false` in `.hhconfig`. We want to remove that in future, so I think this is a good summary.